src/lib/DefaultRuleManager.ts

```diff
@@ -40,27 +49,47 @@ export class DefaultRuleManager implements RuleManager {
 		return this.catalog.getRulesMatchingFilters(filters);
 	}
 
-	async runRulesMatchingCriteria(filters: RuleFilter[], target: string[] | string, format: OUTPUT_FORMAT): Promise<string> {
-		// If target is a string, it means it's an alias for an org, instead of a bunch of local file paths. We can't handle
-		// running rules against an org yet, so we'll just throw an error for now.
-		if (typeof target === 'string') {
-			throw new SfdxError('Running rules against orgs is not yet supported');
-		}
+	async runRulesMatchingCriteria(filters: RuleFilter[], targets: string[], format: OUTPUT_FORMAT): Promise<string | { columns; rows }> {
+		let results: RuleResult[] = [];
 
-		// Convert our filters into paths that we can feed to the engines. If we didn't find any paths, we're done.
-		const paths: PathGroup[] = await this.catalog.getNamedPathsMatchingFilters(filters);
-		if (paths == null || paths.length === 0) {
-			this.logger.trace('No Rule paths found. Nothing to execute.');
-			return '';
+		// Derives rules from our filters to feed the engines.
+		const ruleGroups: RuleGroup[] = await this.catalog.getRuleGroupsMatchingFilters(filters);
+		const rules: Rule[] = await this.catalog.getRulesMatchingFilters(filters);
+		const ps: Promise<RuleResult[]>[] = [];
+		for (const e of this.engines) {
+			const engineGroups = ruleGroups.filter(g => g.engine === e.getName());
+			const engineRules = rules.filter(r => r.engine === e.getName());
+			const engineTargets = await this.unpackTargets(e, targets);
+			ps.push(e.run(engineGroups, engineRules, engineTargets));
 		}
-
-		const ps = this.engines.map(e => e.run(paths, target));
-		const [results] = await Promise.all(ps);
+		const psResults: RuleResult[][] = await Promise.all(ps);
+		psResults.forEach(r => results = results.concat(r));
```

Comment:
A bit more commenting about what's going on here would go a long way.
E.g., it could be clearer that we're concatenating the results from all engines into a single array, and why.
Also, why is `psResults` a nested array?
Also-also, why are we passing in both engineGroups and engineRules?

Reply:
psResults is an array of arrays because each promise returns an array of results.

We are passing in both engine rules and engine groups because the engine may want or need both.  eslint needs the rules array since it doesn't support invoking categories directly.  PMD is the other way around.